packages/synapse-react-client/src/components/DownloadCart/DownloadListTable.tsx

```diff
@@ -1,91 +1,276 @@
 import {
   AvailableFilter,
-  Direction,
   DownloadListItem,
   DownloadListItemResult,
   FilesStatisticsResponse,
   Sort,
   SortField,
 } from '@sage-bionetworks/synapse-types'
-import { Dropdown, Table } from 'react-bootstrap'
-import { calculateFriendlyFileSize } from '../../utils/functions/calculateFriendlyFileSize'
-import { useGetAvailableFilesToDownloadInfinite } from '../../synapse-queries'
-import { useInView } from 'react-intersection-observer'
-import SynapseClient from '../../synapse-client'
+import {
+  CellContext,
+  createColumnHelper,
+  getCoreRowModel,
+  Row,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table'
 import dayjs from 'dayjs'
-import SortIcon from '../../assets/icons/Sort'
-import { useSynapseContext } from '../../utils'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { useErrorHandler } from 'react-error-boundary'
+import {
+  useGetAvailableFilesToDownloadInfinite,
+  useRemoveFilesFromDownloadList,
+} from '../../synapse-queries'
+import { calculateFriendlyFileSize } from '../../utils/functions/calculateFriendlyFileSize'
+import { formatDate } from '../../utils/functions/DateFormatter'
 import { PRODUCTION_ENDPOINT_CONFIG } from '../../utils/functions/getEndpoint'
+import FileEntityDirectDownload from '../DirectDownload/FileEntityDirectDownload'
 import IconSvg from '../IconSvg/IconSvg'
-import { TOOLTIP_DELAY_SHOW } from '../SynapseTable/SynapseTableConstants'
+import { InteractiveCopyIdsIcon } from '../InteractiveCopyIdsIcon'
+import { BlockingLoader, SynapseSpinner } from '../LoadingScreen/LoadingScreen'
 import { SkeletonTable } from '../Skeleton'
+import { TOOLTIP_DELAY_SHOW } from '../SynapseTable/SynapseTableConstants'
+import ColumnHeader from '../TanStackTable/ColumnHeader'
+import StyledTanStackTable from '../TanStackTable/StyledTanStackTable'
 import { displayToast } from '../ToastMessage'
-import DirectProgrammaticDownload from './DirectProgrammaticDownload'
-import { BlockingLoader } from '../LoadingScreen/LoadingScreen'
-import { Tooltip } from '@mui/material'
-import { InteractiveCopyIdsIcon } from '../InteractiveCopyIdsIcon'
-import localizedFormat from 'dayjs/plugin/localizedFormat'
-import FileEntityDirectDownload from '../DirectDownload/FileEntityDirectDownload'
 import { UserBadge } from '../UserCard/UserBadge'
+import DirectProgrammaticDownload from './DirectProgrammaticDownload'
 
 export const TESTING_TRASH_BTN_CLASS = 'TESTING_TRASH_BTN_CLASS'
 
-dayjs.extend(localizedFormat)
+function notifyRemovedFromDownloadList(
+  notificationTitle: string,
+  fileName: string,
+) {
+  displayToast(`${fileName} has been removed from your list.`, 'success', {
+    title: notificationTitle,
+  })
+}
 
-function InteractiveSortIcon(props: {
-  columnSortBy: SortField
-  sort: Sort | undefined
-  setSort: React.Dispatch<React.SetStateAction<Sort | undefined>>
+function RemoveFromDownloadListButton(props: {
```

Comment:
Encapsulate remove from download cart button in component to use `useRemoveFilesFromDownloadList` hook (using `useMutation` under-the-hood)